### PR TITLE
[#1179] feat(build): Make Gravitino build and run against JDK8, 11, 17

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR proposes to add multiple JDK version support for building and running Gravitino server, including JDK8, 11, and 17.

Users could specify the version they want to build in `gradle.properties`, like:

```
jdkVersion = 8

or 

jdkVersion = 11

or 

jdkVersion = 17
```

Or users could specify the version by `./gradlew build -PjdkVersion=8`.

Our generated jars are JDK 8 version compatible, so it can run against Java runtime 8, 11, and 17.

### Why are the changes needed?

To make Gravitino build, test and run against multiple JDK versions.

Fix: #1179 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

This was compiled, tested, run on JDK8, 11, and 17 locally.

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/EntitySerDeFactory.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ public static EntitySerDe createEntitySerDe(String name) {
     String className = ENTITY_SERDES.getOrDefault(name, name);
 
     try {
-      return (EntitySerDe) Class.forName(className).newInstance();
+      return (EntitySerDe) Class.forName(className).getDeclaredConstructor().newInstance();
```
Comment (jerryshao) [inline core/src/main/java/com/datastrato/gravitino/EntitySerDeFactory.java:51]: This is deprecated in Java 11+

File: catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/dyn/DynConstructors.java [hunk 2]
```diff
@@ -253,7 +253,7 @@ public <T> Builder hiddenImpl(Class<T> targetClass, Class<?>... types) {
 
       try {
         Constructor<T> hidden = targetClass.getDeclaredConstructor(types);
-        AccessController.doPrivileged(new MakeAccessible(hidden));
+        java.security.AccessController.doPrivileged(new MakeAccessible(hidden));
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/dyn/DynConstructors.java:256]: `AccessController` is deprecated in JDK17 with no replacement, so currently suppress the warning as a workaround.

**Comment by FANNG1** [comment]
How do we ensure the PRs by other development works for JDK  11 and 17 since GitHub action runs with JDK 8. Do we verify it before a release?

**Comment by jerryshao** [comment]
I will change the CI to support running on JDK11 and 17 in the next PR.

File: gradle.properties [hunk 3]
```diff
@@ -14,3 +14,7 @@ version = 0.4.0-SNAPSHOT
 #sonatype credentials
 SONATYPE_USER = admin
 SONATYPE_PASSWORD = password
+
+#jdkVersion is used to specify the version of JDK to build and test against Gravitino, current
+# supported version is 8, 11, and 17.
+jdkVersion = 8
```
Comment (yuqi1129) [inline gradle.properties:20]: What if the value of `jdkVersion` differs from that of the JDK installed actually?

Comment (jerryshao) [reply gradle.properties:20]: It is the same as before, if the specified `jdkVersion` is not installed locally, it will download automatically by gradle.

Comment (yuqi1129) [reply gradle.properties:20]: I mean if the value of `jdkVersion` is 8 and the JDK you installed is JDK11 or JDK17, so the final behavior is?

Comment (jerryshao) [reply gradle.properties:20]: it will build, test against jdk8, no matter which version you installed in your environment.

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
> I will change the CI to support running on JDK11 and 17 in the next PR.

I'm afraid of CI time bombs, I prefer to test on JDK 11 and 17 before release

**Comment by jerryshao** [comment]
I will make the 

> > > I will change the CI to support running on JDK11 and 17 in the next PR.
> > 
> > 
> > I'm afraid of CI time bombs, I prefer to test on JDK 11 and 17 before release since there're few JDK compatiable issues.
> 
> Agreed, we can test locally and make sure it's okay to use 8, 11, and 17. Adding them to CI may make it difficult to develop in Git Hub.

I will take a try to enable all the CI matrix, if it is too long, I'll see how to handle this.

**Pull request merged**